README.md

```diff
@@ -593,98 +595,89 @@ Service Registry managed by Azure Spring Apps:
 ```
-Then, follow [the steps here](https://learn.microsoft.com/azure/spring-apps/github-actions-key-vault#add-access-policies-for-the-credential) to add access policy for the Service Principal.
 
-In the end, add this service principal as secret named "AZURE_CREDENTIALS" in your forked GitHub repo following [the steps here](https://learn.microsoft.com/azure/spring-apps/how-to-github-actions?pivots=programming-language-java#set-up-github-repository-and-authenticate-1).
+The results in the command line will look like this:
 
-### Customize your workflow
-Finally, edit the workflow file `.github/workflows/action.yml` in your forked repo to fill in the subscription ID, Azure Spring Apps instance name, and Key Vault name that you just created:
-```yml
-env:
-  AZURE_SUBSCRIPTION: subscription-id # customize this
-  SPRING_CLOUD_SERVICE: azure-spring-cloud-name # customize this
-  KEYVAULT: your-keyvault-name # customize this
+```bash
+{
+  "@odata.context": "https://graph.microsoft.com/beta/$metadata#applications('000000000-0000-0000-0000-000000000000')/federatedIdentityCredentials/$entity",
+  "audiences": [
+    "api://AzureADTokenExchange"
+  ],
+  "description": "Testing",
+  "id": "000000000-0000-0000-0000-000000000000",
+  "issuer": "https://token.actions.githubusercontent.com",
+  "name": "github-petclinic-actions",
+  "subject": "repo:Azure-Samples/spring-petclinic-microservices:ref:refs/heads/azure"
+}
 ```
-Once you push this change, you will see GitHub Actions triggered to build and deploy all the apps in the repo to your Azure Spring Apps instance.
-![](./media/automate-deployments-using-github-actions.png)
 
-## Unit-3 - Manage application secrets using Azure KeyVault
+It will look like this in Azure AD portal:
+![Federated Credentials](./media/azuread-github-federated-credential.png)
 
-Use Azure Key Vault to store and load secrets to connect to MySQL database.
+> [!NOTE] It can take few minutes to refresh the federated credentials in Azure AD portal.
 
-### Create Azure Key Vault and store secrets
+### Prepare GitHub Secrets
 
-If you skipped the [Automation step](#automate-deployments-using-github-actions), create an Azure Key Vault and store database connection secrets.
+The actions expects the following secrets to be set in your GitHub repository:
 
-```bash
-    az keyvault create --name ${KEY_VAULT} -g ${RESOURCE_GROUP}
-    export KEY_VAULT_URI=$(az keyvault show --name ${KEY_VAULT} --query 'properties.vaultUri' --output tsv)
-```
+* AZURE_TENANT_ID: The tenant ID of the Azure subscription hosting the Azure Spring Apps instance.
+* AZURE_SUBSCRIPTION_ID: The subscription ID of the Azure subscription hosting the Azure Spring Apps instance.
+* AZURE_CLIENT_ID: The client ID of the service principal created in the previous step.
```

Comment:
```suggestion
* AZURE_CLIENT_ID: The client ID of the  Azure Active Directory appliation created in the previous step.
```

Not sure whether the explanation of the client id would be confusing here since there are multiple service principals or IDs generated before.